app/models/membership.rb

```diff
@@ -0,0 +1,17 @@
+class Membership < ApplicationRecord
+  belongs_to :user
+  belongs_to :group
+  belongs_to :added_by, class_name: "User"
+
+  validate :user_and_group_in_same_organisation
+  validates :user, uniqueness: { scope: :group }
+
+private
+
+  def user_and_group_in_same_organisation
```

Comment:
The commit message says the code was from the Rails generator, but as this is custom it should probably go into a separate commit with the appropriate spec tests.